prometheus-to-sd/config/dynamic_source.go

```diff
@@ -18,8 +19,28 @@ const (
 	nameLabel    = "k8s-app"
 )
 
-// SourceConfigsFromDynamicSources takes pod specifications from the Kubernetes API and maps them to source configs.
-func SourceConfigsFromDynamicSources(gceConfig *GceConfig, sources []flags.Uri) ([]SourceConfig, error) {
+// ChangeType defines the possible types of events.
+type ChangeType string
+
+// Possible change types
+const (
+	Added    ChangeType = "ADDED"
+	Modified ChangeType = "MODIFIED"
+	Deleted  ChangeType = "DELETED"
+	Error    ChangeType = "ERROR"
+)
+
+// SourceConfigChange describes how one of the dynamic sources have changed.
+// Is generated when watch generates a new event.
+type SourceConfigChange struct {
+	ChangeType   ChangeType
+	SourceConfig SourceConfig
+	EventObject  string
+	Err          error
+}
```

Comment:
I think introducing internal events is unnecessary complication.

Reply:
I think this is the correct way to go, I need to send SourceConfig and at least a type, so we need some sort of structure, also handling events in the main is not that correct, and it would depend too much on the dynamic source internal things